### DISCUSSION TITLE
Pin mapping with a XIAO RP2040 and zeptoforth

### DISCUSSION THREAD

**Discussion by mcctuxic** [tabemann/zeptoforth#107] [General]
Hi Travis,

when using a XIAO RP2040 the pin mapping is as follows:
Pin on the board       
    Pin number in zeptoforth
0  26   
1  27
2  28
3  29
4    6
5    7
6    ???
7    1
8    2
9    4
10  3
11  3.3V
12 GND

Pin 6 is always "high" (... :) ) 
Is this pin permanently used by zeptoforth?
An equivalent behaviour I found with a zeptoforth on a Pi Pico W. 
With this board, GP0 is always high.

Is it possible to change this and to use these pins for own purposes
without harm anything?
2
Cheers!
Tuxic

Reedit: The github editor successfully removed all spaces from my little table above.
First column: Pin numbering of the XIAO RP2040
Second column: Pin numbering to access the according pin of the board with zeptoforth.

**Comment by tabemann** [comment]
On Sat, Jun 29, 2024 at 9:06 AM mcctuxic ***@***.***> wrote:

> Hi Travis,
>
> when using a XIAO RP2040 the pin mapping is as follows:
> Pin on the board
> Pin number in zeptoforth
> 0 26
> 1 27
> 2 28
> 3 29
> 4 6
> 5 7
> 6 ???
> 7 1
> 8 2
> 9 4
> 10 3
> 11 3.3V
> 12 GND
>
> Pin 6 is always "high" (... :) )
> Is this pin permanently used by zeptoforth?
> An equivalent behaviour I found with a zeptoforth on a Pi Pico W.
> With this board, GP0 is always high.
>
> Is it possible to change this and to use these pins for own purposes
> without harm anything?
>
> Cheers!
> Tuxic
>
Pin 6 on the XIAO RP2040 is GPIO 0 on the RP2040, which is the TX pin of
UART0, which might explain this. Its behavior will vary depending on
whether you have installed a full or a full_usb build. You will want to
leave it as is if you are using UART0, which will be the case if you are
using a full build, whereas if you are using a full_usb build you can
disable UART0 with 0 UART::DISABLE-UART to free up this pin. Then you can
configure it as you wish. (I presume you are using a full_usb build here.)

Travis

**Comment by mcctuxic** [comment]
Hi Travis,

I am using the full_usb version of Zeptoforth, thats right.
And I am very happy, that only an USB-cable (and an RP2040
board of course) is needed to start programming zeptoforth.

Switching off the UART0 had worked perfectly! I did this on 
my XIAO RP2040 and on my Pi Pico W. Great!

Cheers!
Tuxic
